frontend/src/components/GemMap.vue

```diff
@@ -295,11 +315,21 @@ export default defineComponent({
     hidePopup() {
       this.shouldShowPopup = false;
     },
+
+    pickUpGem() {
+      if (!this.openedInfoWindowGem) {
+        console.error("No gem selected for picking up");
+        return;
+      }
+      this.store.dispatch("pickUpGem", { gemId: this.openedInfoWindowGem.id });
```

Comment:
oh crap, I need to tweak implementation on my side right

Reply:
oh no, how did you implement it? I can pull merge your side first, then I'll tweak the implementation from my end

Reply:
PickupGem takes in a Gem prop, is it the correct implementation? :0 Else I'll fix it!

Reply:
Ok! I modified the implementation so that it works with your end. For now, it only works for green, purple and yellow gems. For other gems, an error will be thrown because their assets cannot be found